client/app/services/file-management.js

```diff
@@ -0,0 +1,25 @@
+import Service, { inject as service } from '@ember/service';
+
+export default class FileManagementService extends Service {
+  @service
+  fileQueue;
+
+  // A hash of all File Managers in the app.
+  // There is one File Manager per editable Package.
+  // The key to access a File Manager is the respective
+  // package ID.
+  fileManagers = {}
```

Comment:
I see, so if the user has multiple forms, there will need to be multiple file managers across those forms. Makes sense!

I worry a little about memory leakage here but with ES6 classes it should be good.

Reply:
Alternatively, what do you think about this: what if we created a file manager on the PASForm model itself? It would be an ancillary part of the model class, manage its own state, and maintain a 1-to-1 mapping with the various forms... that way we don't have to manage a dictionary of references to existing file managers.